tests/tools/tflite_loader/src/tflite_loader.cc

```diff
@@ -264,31 +318,69 @@ int main(const int argc, char **argv)
   std::cout << "[Comparison] TFLite run done!" << std::endl;
 
   // Calculate max difference over all outputs
-  float max_difference = 0.0f;
+  float max_float_difference = 0.0f;
+  bool find_unmatched_output = false;
+
   for (uint32_t out_idx = 0; out_idx < num_outputs; out_idx++)
   {
-    const auto &tflite_output_tensor = interpreter->tensor(interpreter->outputs().at(out_idx));
-    const auto &nnfw_output_tensor = outputs[out_idx];
-
-    if (nnfw_output_tensor.size() != tflite_output_tensor->bytes / sizeof(float))
-      std::cout << "[Comparison] Different size of outputs!" << std::endl;
-    // Check max difference
-    float *tflite_out_ptr = tflite_output_tensor->data.f;
-    for (const auto &nnfw_out : nnfw_output_tensor)
-    {
-      if (std::abs(nnfw_out - *tflite_out_ptr) > max_difference)
-        max_difference = std::abs(nnfw_out - *tflite_out_ptr);
+    nnfw_tensorinfo ti;
+    nnfw_output_tensorinfo(onert_session, out_idx, &ti);
 
-      tflite_out_ptr++;
+    bool matched = true;
+    // Check output tensor values
+
+    const auto &ref_output = interpreter->tensor(interpreter->outputs().at(out_idx))->data;
+    const auto &output = outputs[out_idx];
+
+    switch (ti.dtype)
+    {
+      case NNFW_TYPE_TENSOR_BOOL:
+        matched = compareBuffersExactBool(ref_output.uint8, output, out_idx);
+        break;
+      case NNFW_TYPE_TENSOR_UINT8:
+      case NNFW_TYPE_TENSOR_QUANT8_ASYMM:
+        matched = compareBuffersExact<uint8_t>(ref_output.uint8, output, out_idx);
+        break;
+      case NNFW_TYPE_TENSOR_QUANT8_ASYMM_SIGNED:
+        matched = compareBuffersExact<int8_t>(ref_output.int8, output, out_idx);
+        break;
+      case NNFW_TYPE_TENSOR_INT32:
+        matched = compareBuffersExact<int32_t>(ref_output.i32, output, out_idx);
+        break;
+      case NNFW_TYPE_TENSOR_FLOAT32:
+        // TODO better way for handling FP error?
+        for (uint32_t e = 0; e < num_elems(&ti); e++)
```

Comment:
@hseok-oh (optional) We may compare in the same way of `tflite_run`:

```c++
auto equals = [tolerance](float lhs, float rhs) {
  // NOTE Hybrid approach
  // TODO Allow users to set tolerance for absolute_epsilon_equal
  if (nnfw::misc::fp32::absolute_epsilon_equal(lhs, rhs))
  {
    return true;
  }

  return nnfw::misc::fp32::epsilon_equal(lhs, rhs, tolerance);
};
```

I am not saying `tfflite_run`'s way is best, but would like to use same method.

Reply:
Thanks. I'll try that later.